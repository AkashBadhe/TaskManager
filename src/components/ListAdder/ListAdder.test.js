import React from "react";
import { shallow } from "enzyme";
import { ListAdder } from "./index";

describe("ListAdder", () => {
  const boardId = 1;
  const numberLeft = 5;
  let listAdder = shallow(<ListAdder boardId={boardId} numLeft={numberLeft} />);
  it("Should mount `ListAdder`", () => {
    expect(listAdder).toMatchSnapshot();
  });

  it("Should have `Button` with text", () => {
    expect(listAdder.find("Button").props().text).toBe(
      `Add a new list (${numberLeft})`
    );
  });
});
