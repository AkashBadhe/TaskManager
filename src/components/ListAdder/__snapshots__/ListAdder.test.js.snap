// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListAdder Should mount \`ListAdder\` 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    boardId={1}
    numLeft={5}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "onClick": [Function],
      "text": "Add a new list (5)",
      "variant": "list",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "onClick": [Function],
        "text": "Add a new list (5)",
        "variant": "list",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
